web-server/src/controllers/auth/register.ts

```diff
@@ -1,60 +1,60 @@
 import { Request, Response, NextFunction } from 'express';
-
 import { AppDataSource } from 'orm/data-source';
 import { User } from 'orm/entities/users/User';
-import { CustomError } from 'utils/response/custom-error/CustomError';
-import { ErrorArray } from 'utils/response/custom-error/types';
+import { UserError } from 'utils/Errors';
 
 export const register = async (req: Request, res: Response, next: NextFunction) => {
-  const errors = new ErrorArray();
-
-  const { email, username, password } = req.body;
+  const { email, username, password, passwordConfirm } = req.body;
 
   const userRepository = AppDataSource.getRepository(User);
+  
+  const err: UserError = {};
+
   try {
+
     let user = await userRepository.findOne({ where: { email } });
 
     if (user) {
-      errors.put('email', `User with email ${email} already exists`);
+      err.status = 400;
+      err.email = `User with email "${email}" already exists`
     }
 
     user = await userRepository.findOne({ where: { username } });
 
     if (user) {
-      errors.put('username', `User with username ${username} already exists`);
+      err.status = 400;
+      err.username = `User with username "${username}" already exists`;
     }
 
+    user = new User();
+    user.email = email;
+
     try {
-      user = new User();
-      user.email = email;
-
-      try {
-        user.setUsername(username);
-      } catch (err: unknown) {
-        if (err instanceof CustomError) {
-          errors.extend(err.JSON.errors);
-        } else {
-          const customError = new CustomError(400, 'Raw', 'Error', err, errors);
-          return next(customError);
-        }
+      user.setUsername(username);
+    } catch (e) {
+      if ('username' in e) {
+        err.status = 400;
+        err.username = e.username;
+      } else {
+        err.status = 500;
+        err.raw = 'Internal server error';
```

Comment:
`status == 500` already means Internal Server Error (by convention of all web programmers everywhere). No need to put this string here.